README.md

```diff
@@ -1,39 +1,118 @@
-zimlib
-------
+ZIM library
+=============
 
-The `zimlib` library is the standard implementation of the ZIM
-specification.  It is a library which implements read and write
-methods for ZIM files.
+The ZIM library is the reference implementation for the openZIM file
+format. It's a solution to read and write ZIM file on many systems and
+architectures. More information about the ZIM format and the openZIM
+project at http://www.openzim.org/
 
-Use the zimlib in your own software --- for example, reader
-applications --- to make them ZIM-capable without the need having to
-dig too much into the ZIM file format.
+Disclaimer
+----------
 
-To build:
+This document assumes you have a little knowledge about software
+compilation. If you experience difficulties with the dependencies or
+with the ZIM libary compilation itself, we recommend to have a look to
+[kiwix-build](https://github.com/kiwix/kiwix-build).
+
+Preamble
+--------
+
+Although the ZIM library can be compiled/cross-compiled on/for many
+sytems, the following documentation explains how to do it on POSIX
+ones. It is primarly though for GNU/Linux systems and has been tested
+on recent releases of Ubuntu and Fedora.
+
+Dependencies
+------------
+
+The Kiwix library relies on many third parts software libraries. They
+are prerequisites to the Kiwix library compilation. Following
+libraries need to be available:
+
+* Z ................................................. http://zlib.net/
+(package zlib1g-dev on Ubuntu)
+* Bzip2 ......................................... http://www.bzip.org/
+(package libbz2-dev on Ubuntu)
+* Lzma ...................................... http://tukaani.org/lzma/
+(package lzma-dev on Ubuntu)
+* ICU ................................... http://site.icu-project.org/
+(package libicu-dev on Ubuntu)
+* Xapian ......................................... https://xapian.org/
+(package libxapian-dev on Ubuntu)
+
+These dependencies may or may not be packaged by your operating
+system. They may also be packaged but only in an older version. The
+compilation script will tell you if one of them is missing or too old.
+In the worse case, you will have to download and compile bleeding edge
+version by hand.
+
+If you want to install these dependencies locally, then use the ZIM
+library directory as install prefix.
+
+Environnement
+-------------
+
+The ZIM library builds using [Meson](http://mesonbuild.com/) version
+0.34 or higher. Meson relies itself on Ninja, pkg-config and few other
+compilation tools.
+
+Install first the few common compilation tools:
+* Automake
+* Libtool
+* Virtualenv
+* Pkg-config
+
+Then install Meson itself:
 ```
-./autogen.sh
-./configure
-make
+virtualenv -p python3 ./ # Create virtualenv
+source bin/activate      # Activate the virtualenv
+pip install meson        # Install Meson
```

Comment:
`pip3 install meson`

`meson` use python3, we must use `pip3` to install meson in a python3 environment (this must be change in other readme as well)